src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -138,15 +145,25 @@ public static bool IsPathRooted(string path)
         // only contains whitespace characters an ArgumentException gets thrown.
         public static string GetPathRoot(string path)
         {
-            if (path == null) return null;
+            if (path == null)
+                return null;
+
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
+            return new string(GetPathRoot(path.AsReadOnlySpan()));
+        }
+
+        public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
+        {
+            if (PathInternal.IsEffectivelyEmpty(path))
+                return ReadOnlySpan<char>.Empty;
+
             // Need to return the normalized directory separator
             path = PathInternal.NormalizeDirectorySeparators(path);
```

Comment:
@jkotas was right, we shouldn't normalize separators in the span API. Sorry I missed this.